geowebcache/core/src/main/java/org/geowebcache/GeoWebCacheEnvironment.java

```diff
@@ -0,0 +1,134 @@
+import org.springframework.beans.factory.config.PlaceholderConfigurerSupport;
+import org.springframework.core.Constants;
+import org.springframework.util.PropertyPlaceholderHelper;
+import org.springframework.util.PropertyPlaceholderHelper.PlaceholderResolver;
+
+/**
+ * Utility class uses to process GeoWebCache configuration workflow through external environment variables.
+ * 
+ * This class must be used everytime we need to resolve a configuration placeholder at runtime.
+ * <p>
+ * An instance of this class needs to be registered in spring context as follows.
+ * 
+ * <pre>
+ * <code>
+ *         &lt;bean id="geoWebCacheEnvironment" class="org.geowebcache.GeoWebCacheEnvironment" depends-on="geoWebCacheExtensions"/&gt;
+ * </code>
+ * </pre>
+ * 
+ * It must be a singleton, and must not be loaded lazily. Furthermore, this bean must be loaded before any beans that use it.
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public class GeoWebCacheEnvironment {
+
+    /**
+     * logger
+     */
+    public static Log LOGGER = LogFactory.getLog(GeoWebCacheEnvironment.class);
+
+    private static final Constants constants = new Constants(PlaceholderConfigurerSupport.class);
+
+    /**
+     * Constant set via System Environment in order to instruct GeoWebCache to make use or not of the config placeholders translation.
+     * 
+     * Default to FALSE
+     */
+    public final static boolean ALLOW_ENV_PARAMETRIZATION = Boolean
+            .valueOf(System.getProperty("ALLOW_ENV_PARAMETRIZATION", "false"));
+
+    private static final String nullValue = "null";
+
+    private final PropertyPlaceholderHelper helper = new PropertyPlaceholderHelper(
+            constants.asString("DEFAULT_PLACEHOLDER_PREFIX"),
+            constants.asString("DEFAULT_PLACEHOLDER_SUFFIX"),
+            constants.asString("DEFAULT_VALUE_SEPARATOR"), true);
+
+    private final PlaceholderResolver resolver = new PlaceholderResolver() {
+
+        @Override
+        public String resolvePlaceholder(String placeholderName) {
+            return GeoWebCacheEnvironment.this.resolvePlaceholder(placeholderName);
+        }
+    };
+
+    private Properties props;
+
+    /**
+     * Internal "props" getter method.
+     * 
+     * @return the props
+     */
+    public Properties getProps() {
+        return props;
+    }
+
+    /**
+     * Internal "props" setter method.
+     * 
+     * @param props the props to set
+     */
+    public void setProps(Properties props) {
+        this.props = props;
+    }
+
+    protected String resolvePlaceholder(String placeholder) {
+        String propVal = null;
+        propVal = resolveSystemProperty(placeholder);
+
+        if (props != null && propVal == null) {
+            propVal = props.getProperty(placeholder);
+        }
+
+        return propVal;
+    }
+
+    protected String resolveSystemProperty(String key) {
+        try {
+            String value = System.getProperty(key);
```

Comment:
Same as above, using GeoWebCacheExtensions.getProperty would allow more options to set a property.